src/code/z_sram.c

```diff
@@ -1,25 +1,884 @@
+    { ITEM_STICK,     ITEM_NUT,           ITEM_BOMB,         ITEM_BOW,         ITEM_ARROW_FIRE,  ITEM_DINS_FIRE,
+      ITEM_SLINGSHOT, ITEM_OCARINA_FAIRY, ITEM_BOMBCHU,      ITEM_HOOKSHOT,    ITEM_ARROW_ICE,   ITEM_FARORES_WIND,
+      ITEM_BOOMERANG, ITEM_LENS,          ITEM_BEAN,         ITEM_HAMMER,      ITEM_ARROW_LIGHT, ITEM_NAYRUS_LOVE,
+      ITEM_BOTTLE,    ITEM_POTION_RED,    ITEM_POTION_GREEN, ITEM_POTION_BLUE, ITEM_POCKET_EGG,  ITEM_WEIRD_EGG }, // items
+    { 50, 50, 10, 30, 1, 1, 30, 1, 50, 1, 1, 1, 1, 1, 1, 1 },       // ammo
+    0x7777,                                                         // equipment
+    0x125249,                                                       // upgrades
+    0x1E3FFFF,                                                      // questItems
+    { 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }, // dungeonItems
+    { 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8 },    // dungeonKeys
+    0,                                                              // defenseHearts
+    0,                                                              // gsTokens
+};
+
+u16 sDebugSaveChecksum = 0;
+
+/**
+ *  Initialize debug save. This is also used on the Title Screen
+ *  This Save has a mostly full inventory with 10 hearts and single magic.
+ * 
+ *  Some noteable flags that are set:
+ *  Showed mido sword/shield, met deku tree, deku tree mouth opened, 
+ *  used blue warp in gohmas room, zelda fled castle, light arrow cutscene watched,
+ *  and set water level in water temple to highest level.
+ */
+void Sram_InitDebugSave(void) {
+    SaveContext* temp = &gSaveContext;
+
+    bzero(&SAVE_INFO, INFO_SIZE);
+    gSaveContext.numDays = 0;
+    gSaveContext.unk_18 = 0;
+
+    SAVE_PLAYER_DATA = sDebugSavePlayerData;
+    gSaveContext.equips = sDebugSaveEquips;
+    gSaveContext.inventory = sDebugSaveInventory;
+
+    temp->checksum = sDebugSaveChecksum;
+    gSaveContext.horseData.scene = SCENE_SPOT00;
+    gSaveContext.horseData.pos.x = -1840;
+    gSaveContext.horseData.pos.y = 72;
+    gSaveContext.horseData.pos.z = 5497;
+    gSaveContext.horseData.angle = -0x6AD9;
+    gSaveContext.infTable[0] |= 0x5009;
+    gSaveContext.eventChkInf[0] |= 0x123F;
+    gSaveContext.eventChkInf[8] |= 1;
+    gSaveContext.eventChkInf[12] |= 0x10;
+
+    if (LINK_AGE_IN_YEARS == YEARS_CHILD) {
+        gSaveContext.equips.buttonItems[0] = ITEM_SWORD_KOKIRI;
+        Inventory_ChangeEquipment(EQUIP_SWORD, PLAYER_SWORD_KOKIRI);
+        if (gSaveContext.fileNum == 0xFF) {
+            gSaveContext.equips.buttonItems[1] = ITEM_SLINGSHOT;
+            gSaveContext.equips.cButtonSlots[0] = SLOT_SLINGSHOT;
+            Inventory_ChangeEquipment(EQUIP_SHIELD, PLAYER_SHIELD_DEKU);
+        }
+    }
+
+    gSaveContext.entranceIndex = 0xCD;
+    gSaveContext.magicLevel = 0;
+    gSaveContext.sceneFlags[5].swch = 0x40000000;
+}
+
+#ifdef NON_MATCHING
+// regalloc differences at the end
+/**
```

Comment:
Imo it's better to have function level documentation outside the NON_MATCHING block since it's relevant regardless.